src/utils.ts

```diff
@@ -44,20 +44,32 @@ export const buildRecursive = (
     return `${encodeURIComponent(key + suffix)}=${encodeURIComponent(value)}`
   }
 
-  return validKeys(value)
-    .map((k) => buildRecursive(key, value[k], suffix + '[' + k + ']'))
+  return Object.keys(value)
+    .map((nestedKey) => {
+      const nestedValue = value[nestedKey]
+      if (isNeitherNullNorUndefined(nestedValue)) {
+        return buildRecursive(key, nestedValue, suffix + '[' + nestedKey + ']')
```

Comment:
I expanded the validKeys implementation here instead of making its type more magical. I needed the guarantee that `nestedValue` here is not nullish, and `validKeys` types as it is won't give me it.